File: src/main/java/com/budget/tracker/model/Transaction.java
package com.budget.tracker.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "transactions")
@Getter
@Setter
public class Transaction extends BaseEntity {

    @Column(nullable = false)
    private BigDecimal amount;

    @Column(nullable = false)
    private OffsetDateTime transactionDate;

    @Column(nullable = false)
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(name = "type", nullable = false)
    private TransactionType type;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "account_id", nullable = false)
    private Account account;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "label_id")
    private Label label;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(name = "linked_transfer_id")
    private UUID linkedTransferId;
}

File: frontend/src/types/transaction.ts
import type { Account } from './account';
import type { Category } from './category';
import type { Label } from './label';

export type TransactionType = 'INCOME' | 'EXPENSE' | 'LEND' | 'BORROW';

export interface Transaction {
  id: string;
  amount: number;
  type: TransactionType;
  description?: string;
  transactionDate: string;
  accountId: string;
  account?: Account;
  categoryId?: string;
  category?: Category;
  labelIds?: string[];
  labels?: Label[];
  createdAt: string;
}

export interface CreateTransactionRequest {
  amount: number;
  type: TransactionType;
  description?: string;
  transactionDate: string;
  accountId: string;
  categoryId?: string;
  labelIds?: string[];
}
